### DISCUSSION TITLE
What's the best way to add third-party scripts (GTM)?

### DISCUSSION THREAD

**Discussion by emmanuelsw** [ElMassimo/iles#126] [Q&A] [answered]
Hello!
We are building a pretty big site with Îles and our client requested us to install Google Tag Manager and other third-party scrips, we didn't find a good solution without using a library so we ended up doing something like this:

```html
<component :is="'script'">
  <!-- GTM Script -->
</component>
```

This feels a little bit hacky to me, also we couldn't find a way to add an element right after `<body>`, Is there another way to do this?

Thanks in advance!

**Comment by ElMassimo** [comment, marked as answer]
Hi Emmanuel!

That's one way to do it, I don't see any disadvantages for an îles site.

The "supported way" is to use [_Head_ scripts](https://iles-docs.netlify.app/guide/meta-tags#head-and-meta-tags), as in [this example](https://github.com/ElMassimo/iles/blob/main/docs/src/app.ts#L12):

```ts
# src/app.ts
import { defineApp } from 'iles'

export default defineApp({
  head: {
    script: [
      { src: '<third party script url>', async: true, once: true },
    ],
  },
})
```
